main/lsp/LSPIndexer.cc

```diff
@@ -10,86 +10,24 @@
 using namespace std;
 
 namespace sorbet::realmain::lsp {
-
 namespace {
-
-const core::FileHash &findHash(int id, const vector<core::FileHash> &globalStateHashes,
-                               const UnorderedMap<int, core::FileHash> &overriddingStateHashes) {
-    const auto it = overriddingStateHashes.find(id);
-    if (it == overriddingStateHashes.end()) {
-        return globalStateHashes[id];
-    }
-    return it->second;
-}
-
-bool canTakeFastPathInternal(
```

Comment:
I moved this logic into `canTakeFastPath(changedFiles)`, which ends up being a lot cleaner!